### ISSUE TITLE
problem: [/currency/monero] TypeError: Cannot read property 'map' of undefined

### ISSUE THREAD

**Issue by gazhayes** [Blockrazor/blockrazor#942]
Problem: 
```
Exception in defer callback: TypeError: Cannot read property 'map' of undefined
```

Only happens with Monero, not sure what's going on but we should get to the bottom of it to make sure this isn't something too sinister.

**Comment by AshHimself** [comment]
Taking this.

**Comment by AshHimself** [comment]
@gazhayes - looks like some code was trying to run before data was available. I've just added in some if statements to avoid the error. 

This exchange issue where they are showing ID instead of name, do we have a ticket for that?

**Comment by gazhayes** [comment]
BTW is this done?

**Comment by AshHimself** [comment]
Sorry forgot to comment. Fixed #944
